backend/routers/llm_providers.py

```
@@ -0,0 +1,126 @@
+@router.get("", response_model=list[LlmProviderOut])
+def list_providers(db: Session = Depends(get_db), _=Depends(require_admin)):
+    return get_providers(db)
+
+
+@router.post("", response_model=LlmProviderOut, status_code=201)
+def create(data: LlmProviderCreate, db: Session = Depends(get_db), _=Depends(require_admin)):
+    return create_provider(db, data)
+
+
+@router.put("/reorder", response_model=list[LlmProviderOut])
+def reorder(data: LlmProviderReorder, db: Session = Depends(get_db), _=Depends(require_admin)):
```

Ruff (0.15.12)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)